controllers/ovndbcluster_controller.go

```diff
@@ -617,6 +620,23 @@ func (r *OVNDBClusterReconciler) reconcileNormal(ctx context.Context, instance *
 	return ctrl.Result{}, nil
 }
 
+func getDNSDomain(ctx context.Context, c client.Client, Log logr.Logger) string {
+	DNSDomain := ovnv1.DNSSuffix
+	DNSClusterInfoList := &operatorv1.DNSList{}
+	err := c.List(ctx, DNSClusterInfoList)
+	if err != nil {
+		Log.Info(fmt.Sprintf("Warning: Couldn't retrieve DNS cluster info, using default DNS Suffix: %s", DNSDomain))
+		return DNSDomain
+	}
+	// Using this approach in case CP have multiple domains
+	// also it does not depend on the DNS CR name.
+	// ATM in case of multiple domains will return last one.
+	for _, dns := range DNSClusterInfoList.Items {
+		DNSDomain = dns.Status.ClusterDomain
+	}
```

Comment:
I don't know if it's possible or if it'll be a use case the fact that exists multiple DNS domains (besides `cluster.local`) and how OVN operator should handle it

Reply:
Looking at the Reconcile loop of the openshift dns-operator, it only expects a DNS CR named `Default`, Could modify this chunk of code to get the object by name instead of getting a list and iterating over it.
https://github.com/openshift/cluster-dns-operator/blob/master/pkg/operator/controller/controller.go#L180-L184